CHANGELOG.md

```diff
@@ -19,6 +19,8 @@ Increment the:
   [#2123](https://github.com/open-telemetry/opentelemetry-cpp/pull/2123)
 * [BUILD] Build break with old curl, macro CURL_VERSION_BITS unknown
   [#2102](https://github.com/open-telemetry/opentelemetry-cpp/pull/2102)
+* [BUILD] Allow build shared opentelemetry_proto on non-Windows platform.
```

Comment:
Do you think we should mentioned this PR potential breaking change here in changelog, as we have are introducing a new library?

Reply:
> Do you think we should mentioned this PR potential breaking change here in changelog, as we have are introducing a new library?
Sure, the changelog is added now.

> Just curious the motive of this change - was there any build error with WITH_ABSEIL?

Yes, the related CI job is here https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/4707416429/jobs/8349260349 .